src/Interpreters/Cache/FileCache.cpp

```diff
@@ -170,6 +188,35 @@ void FileCache::assertInitialized() const
 }
 
 void FileCache::initialize()
+{
+    // Prevent initialize() from running twice. This may be caused by two cache disks being created with the same path (see integration/test_filesystem_cache).
+    callOnce(initialize_called, [&] {
+        bool need_to_load_metadata = fs::exists(getBasePath());
+        try
+        {
+            if (!need_to_load_metadata)
+              fs::create_directories(getBasePath());
```

Comment:
```suggestion
            if (!need_to_load_metadata)
                fs::create_directories(getBasePath());
```

Reply:
Applied this (and set indentation to 4 spaces in my editor, sorry)